server/midddleware/auth.go

```diff
@@ -113,10 +123,26 @@ func AuthFunction(ctx context.Context, jwtValidator *validator.Validator, config
 
 	if validatedClaims, ok := validatedToken.(*validator.ValidatedClaims); ok {
 		if customClaims, ok := validatedClaims.CustomClaims.(*CustomClaim); ok {
+			isAdmin := fullMethodNameFound && IsAdminApi(fullMethodName)
+			if isAdmin {
+				// admin api being called, let's check if the user is of admin allowed namespaces
+				if !config.Auth.AdminNamespaces.Contains(customClaims.Namespace.Code) {
+					return ctx, api.Errorf(api.Code_UNAUTHENTICATED, "You are not authorized to perform this admin action")
+				}
+			}
 			log.Debug().Msg("Valid token received")
-			token := &AccessToken{
-				Namespace: customClaims.Namespace.Code,
-				Sub:       validatedClaims.RegisteredClaims.Subject,
+			var token *AccessToken
+			if isAdmin {
+				token = &AccessToken{
+					// set the unknown namespace for administration
+					Namespace: "unknown",
```

Comment:
Why do we need to do this? There is already a check above that means there is an admin namespace passed.

Reply:
discussed and updated to only set `unknown` in case if incoming namespace is `""` with legit token.